src/transformers/utils/generic.py

```diff
@@ -750,3 +750,15 @@ def infer_framework(model_class):
             return "flax"
     else:
         raise TypeError(f"Could not infer framework from class {model_class}.")
+
+
+def safe_int(x):
```

Comment:
Docstrings would be helpful here e.g. for inspecting in IDEs: what does it mean for an int to be safe?

Reply:
Indeed a better name is probably a good idea 😅 I called it `safe_int` in a way to "safely cast some value (which could be a python number or tensor) to an integer in a way that respects tracing"

Reply:
I'll swap with torch.int